### PR TITLE
Display html pages for VS Code Extension documentation, if they exist.

### PR DESCRIPTION
**Pull request by Shanedell**

### CODE REVIEW

**Review by mbeckerle — APPROVED** [review]
+1

**Comment by Shanedell** [comment]
@mbeckerle If you don't mind giving this another look over it would be appreciated. I needed to update the code to expect just a single docx instead of using multiple html files. Then it also downloads the docx file now instead of popping open the html that is why the title is just `Download` under the `Documentation` header.

**Review by mbeckerle — APPROVED** [review]
+1

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,6 @@
 
 target
 .jekyll-cache/
+
+# macos junk files
+.DS_Store
```
Comment (stevedlawrence) [inline .gitignore:20]: Remove this. IDE/OS specific files should not be in a repositories .gitignore file. Git has other methods for ignoring those files.

File: site/_layouts/vscode.html [hunk 2]
```diff
@@ -12,6 +12,16 @@
 {% endunless %}
 
 <div class="row">
+  {% assign filename = "Apache-Daffodil-Extension-for-Visual-Studio-Code-" | append: page.title | append: ".docx" %}
+  {% assign path = "/docs/vscode/" | append: page.title | append: '/' | append: filename %}
+  {% assign exists = site.static_files | where: "path", path %}
+  {% if exists.size != 0 %}
+    <div class="col-md-3 text-right" style="float: right; z-index: 3;">
+      <h3>Documentation</h3>
+      <a href="/docs/vscode/{{ page.title }}/{{ filename }}">Download</a>
```
Comment (stevedlawrence) [inline site/_layouts/vscode.html:21]: Can we just use ``href="{{ path }}"`` here?

File: site/_layouts/vscode.html [hunk 3]
```diff
@@ -12,6 +12,16 @@
 {% endunless %}
 
 <div class="row">
+  {% assign filename = "Apache-Daffodil-Extension-for-Visual-Studio-Code-" | append: page.title | append: ".docx" %}
+  {% assign path = "/docs/vscode/" | append: page.title | append: '/' | append: filename %}
```
Comment (stevedlawrence) [inline site/_layouts/vscode.html:16]: I recently discovered a "capture" macro, which could simplify to this:
```
{% capture path %}/docs/vscode/{{page.title}}/Apache-Daffodil-Extension-for-Visual-Studio-Code-{{page.title}}.docx{% endcapture %}
```

**Review by stevedlawrence — APPROVED** [review]
+1 with some possible cleanup suggesions

**Comment by Shanedell** [comment]
@stevedlawrence My latest commit should address your comments if you want to give them another look over. I added two captures one for each file that would be linked, could probably update the if to check if both exist instead of just docx, how do you feel on that one?

**Comment by stevedlawrence** [comment]
I think it's fine. As it is now, we'll either have both or neither. If that changes we can adjust the site.

**Pull request merged**